source/dofs/dof_tools_constraints.cc

```diff
@@ -1584,12 +1660,152 @@ namespace DoFTools
 
                       case FiniteElementDomination::neither_element_dominates:
                       {
-                        // we don't presently know what exactly to do here.
-                        // it isn't quite clear what exactly we would have
-                        // to do here. sit tight until someone trips over
-                        // the following statement and see what exactly is
-                        // going on
-                        Assert (false, ExcNotImplemented());
+                        // make sure we don't get here twice from each cell
+                        if (processed_p_faces.find(cell->face(face)) != processed_p_faces.end())
+                          break;
```

Comment:
Oh, bummer. Keeping the set of faces around is relatively expensive. I think it would be better to find a tie breaker. For example, at such faces you only deal with it if `current_cell < neighbor_cell` or some such rule.